src/flows/pipes/Notification/ExportTask.tsx

```diff
@@ -0,0 +1,303 @@
+          _acc[_curr.key.name] = _curr.value.location.source
+        }
+
+        return _acc
+      }, acc)
+
+      return acc
+    }, {})
+    const params = remove(
+      ast,
+      node =>
+        node.type === 'OptionStatement' && node.assignment.id.name === 'task'
+    ).reduce((acc, curr) => {
+      // eslint-disable-next-line no-extra-semi
+      ;(curr.assignment?.init?.properties || []).reduce((_acc, _curr) => {
+        if (_curr.key?.name && _curr.value?.location?.source) {
+          _acc[_curr.key.name] = _curr.value.location.source
+        }
+
+        return _acc
+      }, acc)
+
+      return acc
+    }, {})
+
+    const conditions = data.thresholds
+      .map(threshold => THRESHOLD_TYPES[threshold.type].condition(threshold))
+      .join(' and ')
+
+    const newQuery = `
+    import "strings"
+    import "regexp"
+    import "influxdata/influxdb/monitor"
+    import "influxdata/influxdb/schema"
+    import "influxdata/influxdb/secrets"
+    import "experimental"
+    ${ENDPOINT_DEFINITIONS[data.endpoint]?.generateImports()}
+
+    check = {
+    _check_id: "${id}",
+    _check_name: "Notebook Generated Check",
+    _type: "custom",
+    tags: {},
+    }
+    notification = {
+    _notification_rule_id: "${id}",
+    _notification_rule_name: "Notebook Generated Rule",
+    _notification_endpoint_id: "${id}",
+    _notification_endpoint_name: "Notebook Generated Endpoint",
+    }
+
+    task_data = ${format_from_js_file(ast)}
+    trigger = ${conditions}
+    messageFn = (r) => ("${data.message}")
+
+    ${ENDPOINT_DEFINITIONS[data.endpoint]?.generateQuery(data.endpointData)}`
+
+    const newAST = parse(newQuery)
+
+    if (!params.name) {
+      params.name = `"Notebook Task for ${id}"`
+    }
+
+    if (data.interval) {
+      params.every = data.interval
+    }
+
+    if (data.offset) {
+      params.offset = data.offset
+    }
+
+    if (Object.keys(vars).length) {
+      const varString = Object.entries(vars)
+        .map(([key, val]) => `${key}: ${val}`)
+        .join(',\n')
+      const varHeader = parse(`option v = {${varString}}\n`)
+      newAST.body.unshift(varHeader.body[0])
+    }
+
+    const paramString = Object.entries(params)
+      .map(([key, val]) => `${key}: ${val}`)
+      .join(',\n')
+    const taskHeader = parse(`option task = {${paramString}}\n`)
+    newAST.body.unshift(taskHeader.body[0])
+
+    return format_from_js_file(newAST)
+  }, [
+    id,
+    queryText,
+    data.every,
+    data.offset,
+    data.endpointData,
+    data.endpoint,
+    data.thresholds,
+    data.message,
+  ])
+
+  const generateTask = useCallback(() => {
+    event('Alert Panel (Notebooks) - Export Alert Task Clicked')
+
+    if (data.thresholds[0].type === deadmanType) {
+      return generateDeadmanTask()
+    } else {
+      return generateThresholdTask()
+    }
+  }, [generateDeadmanTask, generateThresholdTask, data.thresholds])
+
+  const validateTask = async (queryText: string): Promise<boolean> => {
```

Comment:
straddling this validation between the two components is weird. and all essentially for an event name to stay collocated in a button press? feels like the utility of sharing that component has been stretched to it's breaking point here. thoughts on breaking it up here?

Reply:
we spoke offline.
This is a good compromise on what we have now...with the correct solution not being possible at this time.